tests/std/tests/P0896R4_ranges_alg_nth_element/test.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::nth_element(borrowed<false>{}, nullptr_to<int>)), ranges::dangling>);
+STATIC_ASSERT(same_as<decltype(ranges::nth_element(borrowed<true>{}, nullptr_to<int>)), int*>);
+
+using P = pair<int, int>;
+
+struct instantiator {
+    static constexpr int keys[] = {7, 6, 5, 4, 3, 2, 1, 0};
+
+    template <ranges::random_access_range R>
+    static constexpr void call() {
+#if !defined(__clang__) && !defined(__EDG__) // TRANSITION, VSO-938163
+#pragma warning(suppress : 4127) // conditional expression is constant
+        if (!ranges::contiguous_range<R> || !is_constant_evaluated())
+#endif // TRANSITION, VSO-938163
+        {
+            using ranges::nth_element, ranges::all_of, ranges::find, ranges::iterator_t, ranges::less, ranges::none_of,
+                ranges::size;
+
+            P input[size(keys)];
+            const auto init = [&] {
+                for (size_t j = 0; j < size(keys); ++j) {
+                    input[j] = P(keys[j], static_cast<int>(10 + j));

```

Comment:
Use braces to construct `P` for consistency with the code below?

Reply:
Ha! I changed this from `{}` to ()` to avoid narrowing, and then added the `static_cast` to silence narrowing conversion warnings.